ledger/store/src/helpers/rocksdb/internal/mod.rs

```diff
@@ -189,6 +193,30 @@ impl Database for RocksDB {
 }
 
 impl RocksDB {
+    /// Toggles the atomic override; if it becomes disabled, the pending
+    /// operations get executed.
+    fn flip_atomic_override(&self) -> Result<bool> {
+        // The override is only intended to be toggled before or after
+        // atomic batches - never in the middle of them.
+        assert_eq!(self.atomic_depth.load(Ordering::SeqCst), 0);
+
+        // https://github.com/rust-lang/rust/issues/98485
+        let previous_value = self.atomic_override.load(Ordering::SeqCst);
+
+        // A flip from enabled to disabled executes all pending operations
+        // as a single atomic batch.
+        if previous_value {
+            let batch = mem::take(&mut *self.atomic_batch.lock());
+            self.rocksdb.write(batch)?;
```

Comment:
Why do we need to write all pending operations when flipping? Please provide commentary in the code for this.

Reply:
I added some extra comments; there could be an additional operation responsible for performing them, but that would require a new method that would most likely be called at basically the same spot. Since the purpose of this solution is to merge 2 atomic batches into 1, it makes sense to do it automatically at the end. The overall order of operations is as follows
1. enable (flip from disabled to enabled) the atomic override, making the subsequent atomic batches only collect operations, but not execute them
2. execute block insertion and finalization
3. disable (flip from enabled to disabled) the atomic override, executing all the collected operations since it has been enabled (or discarding them in case of failure)